File: frontend/src/app/services/models/task-response-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ProjectResponseDto } from '../models/project-response-dto';
import { UserResponseDto } from '../models/user-response-dto';
export interface TaskResponseDto {
  deadline?: string;
  description?: string;
  id?: number;
  name?: string;
  priority?: 'LOW' | 'MEDIUM' | 'HIGH';
  project?: ProjectResponseDto;
  responsibleUser?: UserResponseDto;
  status?: 'NOT_STARTED' | 'IN_PROGRESS' | 'COMPLETED';
}

File: src/main/java/bohdan/markatov/org/dto/task/TaskResponseDto.java
package bohdan.markatov.org.dto.task;

import bohdan.markatov.org.dto.project.ProjectResponseDto;
import bohdan.markatov.org.dto.user.UserResponseDto;
import bohdan.markatov.org.model.Task;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class TaskResponseDto {
    private Long id;
    private String name;
    private String description;
    private Task.Status status;
    private Task.Priority priority;
    private ProjectResponseDto project;
    private UserResponseDto responsibleUser;
    private LocalDateTime deadline;
}
